CorsixTH/Lua/entities/object.lua

```diff
@@ -651,9 +657,10 @@ function Object:onClick(ui, button, data)
       if self.object_type.class == "Machine" then
         taskType = "repairing"
       end
-      local index = self.hospital:getIndexOfTask(self.tile_x, self.tile_y, taskType)
-      if index ~= -1 then
-        self.hospital:removeHandymanTask(index, taskType)
+      local task_index = self.hospital:getIndexOfTask(self.tile_x, self.tile_y, taskType)
+      local task_exist = task_index ~= -1
+      if task_exist then
```

Comment:
You can also do the test in the `if` condition, and write a comment behind it what it does.

Reply:
Here you use additional code to explain WHAT it does, rather than just do things.

To explain what it does text often wins, but it's a trade-off.
Different people have different ideas about the best balance is.

That is why I stated it as an alternative, so you are aware of it. Whether you follow it, is up to you.